src/SNLS_device_forall.h

```diff
@@ -126,9 +251,10 @@ namespace snls {
    /// limitation of this wrapper is that the lambda captures can
    /// only capture functions / variables that are publically available
    /// if this is called within a class object.
-   template <const int NUMTHREADS, typename DBODY, typename HBODY>
-   inline void SNLS_ForallWrap(const int st,
```

Comment:
No inline for the same reasons as above (and everywhere else in this file).  The only cases you really need it is if these functions are partial/full specializations (again because of the ODR rule), which they are not here.